data-structures/src/data/structure/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,239 @@
+        requestedPlace.data = element;
+
+        return replacedElement;
+    }
+
+    public void clear() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    public E remove(int index) {
+        if (index >= size && index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        Node requestedPlace = findNode(index);
+
+        if (size == 1) {
+            clear();
+            return requestedPlace.data;
+        }
+
+        if (index == 0) {
+            requestedPlace.next.previous = null;
+            head = requestedPlace.next;
+            size--;
+            return requestedPlace.data;
+        }
+
+        if (index == size - 1) {
+            requestedPlace.previous.next = null;
+            tail = requestedPlace.previous;
+            size--;
+            return requestedPlace.data;
+        }
+
+        requestedPlace.previous.next = requestedPlace.next;
+        requestedPlace.next.previous = requestedPlace.previous;
+        size--;
+        return requestedPlace.data;
+    }
+
+    public boolean contains(E element) {
+        requireNonNull(element, "The element must not be null!");
+
+        Node temp = head;
+
+        for (int i = 0; i < size; i++) {
+            if (temp.data.equals(element)) {
+                return true;
+            }
+
+            temp = temp.next;
+        }
+
+        return false;
+    }
+
+    public int indexOf(E element) {
+        requireNonNull(element, "The element must not be null!");
+
+        Node temp = head;
+
+        for (int i = 0; i < size; i++) {
+            if (temp.data.equals(element)) {
+                return i;
+            }
+
+            temp = temp.next;
+        }
+
+        return -1;
+    }
+
+    public int size() {
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public E get(int index) {
+        if (index >= size || index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        return findNode(index).data;
+    }
+
+    private LinkedList<E> makeCopyList(final LinkedList<E> other) {
+        requireNonNull(other, "The list must not be null!");
+
+        if (other.isEmpty()) {
+            throw new IllegalArgumentException("The list must not be empty!");
+        }
+
+        final LinkedList<E> copy = new LinkedList<>();
+        Node temp = other.head;
+
+        for (int i = 0; i < other.size; i++) {
+            copy.add(temp.data);
+            temp = temp.next;
+        }
+
+        return copy;
+    }
+
+    private Node findNode(int index) {
+        if (index >= size || index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        Node requestedNode = head;
+
+        for (int i = 0; i < index; i++) {
+            requestedNode = requestedNode.next;
+        }
+
+        return requestedNode;
+    }

```

Comment:
These two methods are key! It's a great thing that you extracted this logic into separate methods. Makes the code much easier to read and maintain. 🥇